dev/com.ibm.ws.jdbc_fat/test-applications/basicfat/src/basicfat/DataSourceTestServlet.java

```diff
@@ -2250,6 +2253,177 @@ public void testOnErrorWARN() throws Exception {
+     * Note, onError=IGNORE will have the same data source behavior as WARN,
+     * but will not log an error.
+     */
+    public void testOnErrorWorkingTwoDSorderA() throws Exception {
+
+        DataSource dsfat11bmixedpoolderby = null;
+        DataSource dsfat11amixedpoolderby = null;
+
+        String firstLookupJNDIName = "jdbc/dsfat11amixedpoolderby";
+        String secondLookupJNDIName = "jdbc/dsfat11bmixedpoolderby";
+
+        String firstLookupCon = null;
+        String secondLookupCon = null;
+
+        for (int i = 0; i < 2; i++) {
+            tran.setTransactionTimeout(60);
+            try {
+                tran.begin();
+                try {
+                    dsfat11bmixedpoolderby = (DataSource) new InitialContext().lookup(firstLookupJNDIName);
+                    Connection con = dsfat11bmixedpoolderby.getConnection();
+                    String pool = showPoolContents(firstLookupJNDIName);
+                    assertFalse("Connection pool should exist for jndi name " + firstLookupJNDIName, pool.equals("Not found"));
+
+                    if (firstLookupCon == null) {
+                        String indexString = "WSRdbManagedConnectionImpl@";
+                        int index = pool.indexOf(indexString);
+                        int endindex = index + indexString.length() + 10;
+                        firstLookupCon = pool.substring(index, endindex);
+                    } else {
+                        assertTrue("Did not find expected connection " + firstLookupCon + " in pool " + pool, pool.contains(firstLookupCon));
+                    }
+
+                    // extra check for data source being used, should be 14
+                    Statement stmt = con.createStatement();
+                    int timeout = stmt.getQueryTimeout();
+                    assertTrue("Incorrect query timeout, expecting 14 but found " + timeout, timeout == 14);
+                    con.close();
+                } finally {
+                    tran.commit();
+                }
+            } finally {
+                tran.setTransactionTimeout(0); // restore default
+            }
+
+            tran.setTransactionTimeout(60);
+            try {
+                tran.begin();
+                try {
+                    dsfat11amixedpoolderby = (DataSource) new InitialContext().lookup(secondLookupJNDIName);
+                    Connection con = dsfat11amixedpoolderby.getConnection();
+                    String pool = showPoolContents(firstLookupJNDIName);
+                    assertFalse("Connection pool should exist for jndi name " + firstLookupJNDIName, pool.equals("Not found"));
+
+                    if (secondLookupCon == null) {
+                        String indexString = "WSRdbManagedConnectionImpl@";
+                        int index = pool.indexOf(indexString);
+                        int endindex = index + indexString.length() + 10;
+                        secondLookupCon = pool.substring(index, endindex);
```

Comment:
It seems like secondLookupCon will end up with the same value as firstLookupCon because the steps prior to this look like an exact copy.